packages/web/src/views/launchpad/launchpad-detail/components/launchpad-my-participation/LaunchpadMyParticipation.tsx

```diff
@@ -0,0 +1,178 @@
+import { ProjectRewardInfoModel } from "../../LaunchpadDetail";
+import { useGetLastedBlockHeight } from "@query/pools";
+import { LAUNCHPAD_REFETCH_INTERVAL } from "@common/values";
+
+import { MyParticipationWrapper } from "./LaunchpadMyParticipation.styles";
+import Button, { ButtonHierarchy } from "@components/common/button/Button";
+import LaunchpadMyParticipationBox from "./launchpad-my-participation-box/LaunchpadMyParticipationBox";
+import { useLaunchpadClaimAllModal } from "../../hooks/use-launchpad-claim-all-modal";
+import LaunchpadMyParticipationUnconnected from "./launchpad-my-participation-unconnected/LaunchpadMyParticipationUnconnected";
+import LaunchpadMyParticipationNoData from "./launchpad-my-participation-no-data/LaunchpadMyParticipationNoData";
+import { toNumberFormat } from "@utils/number-utils";
+import LaunchpadMyParticipationSkeleton from "./launchpad-my-participation-skeleton/LaunchpadMyParticipationSkeleton";
+
+interface LaunchpadMyParticipationProps {
+  poolInfos: LaunchpadPoolModel[];
+  data: LaunchpadParticipationModel[];
+  rewardInfo: ProjectRewardInfoModel;
+  connected: boolean;
+  isSwitchNetwork: boolean;
+  isFetched: boolean;
+  isLoading: boolean;
+  status: string;
+
+  refetch: () => Promise<void>;
+}
+
+const LaunchpadMyParticipation = ({
+  poolInfos,
+  data,
+  rewardInfo,
+  connected,
+  isSwitchNetwork,
+  isFetched,
+  isLoading,
+  status,
+  refetch,
+}: LaunchpadMyParticipationProps) => {
+  const { claim } = useLaunchpadHandler();
+  const { openLaunchpadClaimAllModal } = useLaunchpadClaimAllModal({
+    data,
+    rewardInfo,
+    refetch,
+  });
+
+  const handleClickClaim = React.useCallback(
+    (data: LaunchpadParticipationModel) => {
+      claim(data, async () => {
+        await refetch();
+      });
+    },
+    [claim, refetch],
+  );
+
+  const handleClickClaimAll = React.useCallback(() => {
+    openLaunchpadClaimAllModal();
+  }, [openLaunchpadClaimAllModal]);
+
+  const highestApr = React.useMemo(() => {
+    return poolInfos.reduce((acc, current) => {
+      if (Number(current.apr) > acc) {
+        return Number(current.apr);
+      }
+      return acc;
+    }, Number(poolInfos?.[0]?.apr ?? 0));
+  }, [poolInfos]);
+
+  const { data: blockHeight } = useGetLastedBlockHeight({
+    refetchInterval: LAUNCHPAD_REFETCH_INTERVAL,
+  });
+
+  const isShowClaimAllButton = React.useMemo(() => {
+    if (!blockHeight) return;
+
+    const currentBlockHeight = blockHeight;
+
+    return data.some(item => {
+      const isClaimableBlockHeight = BigNumber(
+        currentBlockHeight,
+      ).isGreaterThan(item.claimableBlockHeight);
+
+      const isClaimedReward =
+        Number(toNumberFormat(item.claimableRewardAmount, 2)) === 0;
+      const isClaimedDeposit = Number(toNumberFormat(item.depositAmount)) === 0;
+      const isClaimed = isClaimedReward && isClaimedDeposit;
+
+      return !isClaimed && isClaimableBlockHeight;
+    });
+  }, [data, blockHeight]);
+
```

Comment:
The `blockHeight` dependency should be added.

Reply:
`blockHeight` is already in the dependency, did he mean something else?

Reply:
I was wrong, it's working fine.